docs/performance.md

```diff
@@ -1,194 +1,88 @@
 
-1. Create a [GCP free tier](https://cloud.google.com/free/docs/free-cloud-features#free-tier) e2-micro VM instance in us-central1 (iowa).
+### Resources
 
-1. [Install Go](https://go.dev/doc/install)
-   
-   ```sh
-   instance:~$ wget https://go.dev/dl/go1.23.0.linux-amd64.tar.gz
-   instance:~$ sudo rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go1.23.0.linux-amd64.tar.gz
-   instance:~$ export PATH=$PATH:/usr/local/go/bin
-   instance:~$ go version
-   go version go1.23.0 linux/amd64
-   ```
+Exact performance numbers are highly dependent on the exact infrastructure being used (e.g. storage type & locality, host resources
+of the machine(s) running the personality binary, network speed and weather, etc.)  If in doubt, you should run your own performance
+tests on infrastructure which is as close as possible to that which will ultimately be used to run the log in production.
+The [conformance binaries](/cmd/conformance) and [hammer tool](/internal/hammer) may be helpful for running such tests.
 
-1. [Install Docker using the `apt` repository](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository)
+### Deduplication
 
-1. Install Git
+Deduplicating incoming entries is a somewhat expensive operation in terms of both storage and throughput.
+Not all personality designs will require it, so Tessera is build such that you only incur these costs if they are necessary
+for your design.
 
-   ```sh
-   instance:~$ sudo apt-get install git -y -q
-   ...
-   instance:~$ git version
-   git version 2.39.2
-   ```
+Leaving deduplication disabled will greatly increase the throughput of the log, and decrease CPU and storage costs.
 
-1. Clone the Trillian Tessera repository
 
-   ```sh
-   instance:~$ git clone https://github.com/transparency-dev/trillian-tessera.git
-   Cloning into 'trillian-tessera'...
-   ```
+## Backends
 
-1. Run `cmd/conformance/mysql` and MySQL database via Docker compose
+The currently supported storage backends are listed below, with a rough idea of the expected performance figures.
+Individual storage implementations may have more detailed information about performance in their respective directories.
 
-   ```sh
-   instance:~/trillian-tessera$ sudo docker compose -f ./cmd/conformance/mysql/docker/compose.yaml up
-   ```
+### GCP
 
-1. Run `hammer` and get performance metrics
+The main lever for cost vs performance on GCP is Spanner, in the form of "Performance Units" (PUs).
+PUs can be allocated in blocks of 100, and 1000 PUs is equivalent to 1 Spanner Server.
 
-   ```sh
-   hammer:~/trillian-tessera$ go run ./internal/hammer --log_public_key=transparency.dev/tessera/example+ae330e15+ASf4/L1zE859VqlfQgGzKy34l91Gl8W6wfwp+vKP62DW --log_url=http://10.128.0.3:2024 --max_read_ops=0 --num_writers=512 --max_write_ops=512
-   ```
+The table below shows some rough numbers of measured performance:
 
-### GCP Free Tier VM Instance + Cloud SQL (MySQL)
+| Spanner PUs | Num FEs | QPS no-dedup | QPS dedup |
```

Comment:
It's a bit odd that we're publishing the number for dedup when this code isn't availale for usage right now?

Reply:
I'll raise a PR with that code so folks can reproduce.